eth_abi/decoding.py

```diff
@@ -340,6 +341,74 @@ def from_type_str(cls, abi_type, registry):
         return cls.as_decoder(value_bit_size=abi_type.sub * 8)
 
 
+class BaseFixedDecoder(Fixed32ByteSizeDecoder):
+    frac_places = None
+    is_big_endian = True
+
+    @classmethod
+    def validate(cls):
+        super().validate()
+
+        if cls.frac_places is None:
+            raise ValueError("must specify `frac_places`")
+
+        if not (cls.frac_places > 0 or cls.frac_places <= 80):
+            raise ValueError("`frac_places` must be in range (0, 80]")
+
+
+class UnsignedFixedDecoder(BaseFixedDecoder):
+    @classmethod
+    def decoder_fn(cls, data):
+        value = big_endian_to_int(data)
+
+        with decimal.localcontext(abi_decimal_context):
+            decimal_value = decimal.Decimal(value) * TEN ** -cls.frac_places
```

Comment:
I know they are equal, but I expected this to be `decimal.Decimal(value) / TEN ** cls.frac_places`.  Anyone else find that more intuitive or semantically correct?

Reply:
Yes, that's also what I would have expected. It's a bit more visually apparent what's happening. Although some of it is that I don't read `-variable` as smoothly as I read `-1 * variable`.